be/src/olap/in_list_predicate.h

```diff
@@ -76,33 +76,55 @@ namespace doris {
 class VectorizedRowBatch;
 
 // todo(wb) support evaluate_and,evaluate_or
+template <class T>
+class InListPredicate : public ColumnPredicate {
+public:
+    InListPredicate(uint32_t column_id, phmap::flat_hash_set<T>&& values, bool in_or_not, bool is_opposite = false);
+    PredicateType type() const override { return PredicateType::IN_LIST; }
+    virtual void evaluate(VectorizedRowBatch* batch) const override;
+    void evaluate(ColumnBlock* block, uint16_t* sel, uint16_t* size) const override;
+    void evaluate_or(ColumnBlock* block, uint16_t* sel, uint16_t size,
+                        bool* flags) const override;
+    void evaluate_and(ColumnBlock* block, uint16_t* sel, uint16_t size,
+                        bool* flags) const override;
+    virtual Status evaluate(const Schema& schema,
+                            const std::vector<BitmapIndexIterator*>& iterators,
+                            uint32_t num_rows, roaring::Roaring* bitmap) const override;
+    void evaluate(vectorized::IColumn& column, uint16_t* sel, uint16_t* size) const override;
+    void evaluate_and(vectorized::IColumn& column, uint16_t* sel, uint16_t size,
+                        bool* flags) const override {}
+    void evaluate_or(vectorized::IColumn& column, uint16_t* sel, uint16_t size,
+                        bool* flags) const override {}
+
+private:
+    phmap::flat_hash_set<T> _values;
+    bool _in_or_not; //true for in(...), false for not in (...)
+};
+
+// template <class T>
```

Comment:
delete these commented codes?

Reply:
It is better remove these codes when we clean non-vectorized code.

Reply:
The NotInListPredicate definition is removed.